File: Algorithms/Easy/P1-Two_Sum/p1.ac.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        newlist = nums.copy()
        newlist.sort()
        front = 0
        end = len(newlist) - 1
        while True:
            if newlist[front] + newlist[end] < target:
                front += 1
            if newlist[front] + newlist[end] > target:
                end -= 1
            if newlist[front] + newlist[end] == target:
                break
        ans = []
        for i in range(0, len(nums)):
            if nums[i] == newlist[front]:
                ans.append(i)
            elif nums[i] == newlist[end]:
                ans.append(i)
        return ans

File: Algorithms/Easy/P1-Two_Sum/p1.ac.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut v = Vec::<i32>::new();
        for i in &nums {
            v.push(*i);
        }
        v.sort();
        let (mut front, mut end): (usize, usize) = (0, v.len() - 1);
        loop {
            if (front > end || v[front] + v[end] == target) {
                break;
            } else if v[front] + v[end] > target {
                end -= 1;
            } else if v[front] + v[end] < target {
                front += 1;
            }
        }
        let (front, end) = (v[front], v[end]);
        let mut r = Vec::<i32>::new();
        for i in 0..nums.len() {
            if &nums[i] == &front {
                r.push(i as i32);
            } else if &nums[i] == &end {
                r.push(i as i32);
            }
        }
        r
    }
}
